src/Seabug/Contract/CnftMint.purs

```diff
@@ -0,0 +1,85 @@
+module Seabug.Contract.CnftMint where
+
+import Contract.Prelude
+
+import Contract.Address
+  ( getNetworkId
+  , ownPaymentPubKeyHash
+  , ownStakePubKeyHash
+  , payPubKeyHashBaseAddress
+  )
+import Contract.AuxiliaryData (setTxMetadata)
+import Contract.Monad (Contract, liftContractM, liftedE, liftedM)
+import Contract.Prim.ByteArray (hexToByteArray)
+import Contract.ScriptLookups as Lookups
+import Contract.Scripts (mintingPolicyHash)
+import Contract.Transaction (TransactionHash, balanceAndSignTxE, submit)
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (utxosAt)
+import Contract.Value
+  ( CurrencySymbol
+  , TokenName
+  , mkTokenName
+  , scriptCurrencySymbol
+  , singleton
+  )
+import Data.Array (head)
+import Data.Map as Map
+import Data.NonEmpty as NonEmpty
+import Metadata.Cip25 (Cip25Metadata(..), Cip25MetadataEntry(..))
+import Seabug.CnftMintPolicy (mkCnftMintingPolicy)
+import Seabug.Types (MintCnftParams(..))
+
+-- | Mint a collection NFT
+mintCnft
+  :: forall (r :: Row Type)
+   . MintCnftParams
+  -> Contract r (TransactionHash /\ (CurrencySymbol /\ TokenName))
+mintCnft (MintCnftParams params) = do
+  owner <- liftedM "Cannot get PaymentPubKeyHash" ownPaymentPubKeyHash
+  ownerStake <- liftedM "Cannot get StakePubKeyHash" ownStakePubKeyHash
+  networkId <- getNetworkId
+  addr <- liftContractM "Cannot get user address" $
+    payPubKeyHashBaseAddress networkId owner ownerStake
+  utxos <- liftedM "Cannot get user utxos" $ utxosAt addr
+  oref /\ _ <- liftContractM "Cannot find user utxo"
+    $ head
+    $ Map.toUnfoldableUnordered (unwrap utxos)
+  policy <- liftedE $ mkCnftMintingPolicy oref
+  curr <- liftedM "Could not get currency symbol" $ liftAff $
+    scriptCurrencySymbol policy
+  -- TODO: figure out how to encode the token name (base64 maybe)
```

Comment:
Did we come to a conclusion here?

Reply:
The relevant issue is #26 , for now it's not necessary to encode it as we do the minting manually

Reply:
Right